fip-6.md

```diff
@@ -0,0 +1,376 @@
+* verify that the fee for this does not exceed the max fee specified.
+* charge appropriate fee (this will be a bundled fee transaction, fee will be based on number of periods specified)
+* create entry in the locktokens table for these lock tokens.
+* verify tx does not exceed max transaction size.
+* increase RAM limit by 200+(60*number of lock periods) bytes
+* Return status json containing status (ok), and fee charged.
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|Invalid receiver public key|Invalid format|400|"receiver_public_key"|value of public key|"Invalid Public Key"|
+|Invalid votable|Invalid format|400|"votable"|votable invalid|"Invalid votable value"|
+|Lock Periods Invalid|less of equal to 0 duration specified on any lock period, or total percent does not sum to 100.0|400|"lock_periods"|Invalid Lock Periods |
+|Invalid Actor|Actor does not exist on chain|403|||"Invalid Actor"|
+|Insufficient balance| Actor does not have required amount of funds|400|"amount"|Value sent in, e.g. "100"|"Insufficient balance"|
+|Fee exceeds maximum|Actual fee is greater than supplied max_fee|400|max_fee"|Value sent in, e.g. "1000000000"|"Fee exceeds supplied maximum"|
+##### Response
+|Group|Parameter|Format|Definition|
+|---|---|---|---|
+||status|String|Ok|
+||fee_collected|String|fee amount collected SUFs|
+|lock_id| string| the id of the lock that was created
+###### Example
+```
+{
+  "lock_id": 5
+  "status": "Ok",
+  "fee_collected": 0		
+}
+```
+## Fees
+a new fee will be created for lock_tokens, 600000000 SUF per lock period, this is a mandatory fee.
+
+## Specification
+### UnLock tokens
+#### New end point: *unlock_tokens* 
+#### New action in new fio.lock contract unlocktoks
+##### Request
+When this request is processed, inputs will be verified,  the specified locks locking periods will be processed and the lock witll be updated,
+then the system will unlock the specified amount of tokens from the specified lock if the lock has the specified amount of tokens available in the remaining lock amount. 
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|lock_id|Yes| int64 id of the lock .|the lock upon which to perform unlocking.|
+|amount|Yes|SUF amount|amount of FIO to unlock|
+|max_fee|Yes|max fee SUFs|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by /get_fee for correct value.|
+|actor|Yes|FIO account name|FIO account for the signer|
+###### Example
+```
+{
+  "lock_id": 1,
+  "actor": "aftyershcu22",
+  "amount": 400000000000
+  "max_fee": 40000000000
+}
+```
+##### Processing
+* require auth of the actor
+* Verify format and values for lock_id, amount, max_fee, actor. invalid input error if incorrect.
+* Verify the lock id exists in the locktokens table, error if none exist.
+* Verify the owner of the lock matches the actor. signature error if no match
```

Comment:
Who is the owner of the lock, the actor who created it or the actor whose account is locked?
There is no "signature error" in Exception handling.

Reply:
the actor whose account is locked.

Reply:
might it be better to merge this into the FIPs repo, then have pawel edit in the desired changes..some of these can get mis-interpreted causing extra iterations...